packages/grafana-data/src/types/datasource.ts

```diff
@@ -351,6 +351,19 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /*
+   * Optionally, use this method to track information about queries.
+   * Only executed once, during the dashboard initialization.
+   * @alpha -- experimental
+   */
+  trackQueries?(
```

Comment:
**Suggestion:** I would probably call this something like `onTrackQuery()` instead to follow what we have in our [FE style guide](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md#name-callback-props-and-handlers-with-an-on-prefix)

I think with this naming it is also easier to understand on the call-site that this is actually a callback that will be called by the runtime.

Reply:
Thinking about this a bit more and after talking to @academo I think I was probably wrong here, and `trackQueries()` is actually fine. (I was probably tricked by the fact that this is a method designed to be called only by the runtime, kind of as like a "callback".